### ISSUE TITLE
Bug: Beatles Radio stops after awhile

### ISSUE THREAD

**Issue by linknum23** [micro-nova/AmpliPi#165]
Need to investigate what is happening.
I have started investigating by running the following command on my ubuntu laptop:
    python streams/runvlc.py http://www.beatlesradio.com:8000/stream/1/

I noticed that during some advertisements something changes with the stream and causes this error to print out:
```
Warning: Big change (MPEG version, layer, rate). Frankenstein stream?
[00007f871404ec10] main decoder error: Timestamp conversion failed (delay 1000000, buffering 4500, bound 3000000)
[00007f871404ec10] main decoder error: Could not convert timestamp 1246137212823 for mpg123
[00007f871404ec10] main decoder error: Timestamp conversion failed (delay 1000000, buffering 4500, bound 3000000)
```

It eventually recovered:
```
...
[00007f871404ec10] main decoder error: Timestamp conversion failed (delay 1000000, buffering 100000, bound 3000000)
[00007f871404ec10] main decoder error: Could not convert timestamp 1246167308220 for mpg123
[0000000001d39320] main input error: ES_OUT_SET_(GROUP_)PCR  is called too late (pts_delay increased to 1000 ms)
Current track: Can't Buy Me Love - The Beatles
```

I wonder if it is able to recover on the pi, testing now. We may need to add a way to restart the vlc stream if it fails

Workaround:

1. Select the empty stream (or any other stream)
2. Select _Beatles Radio_

This sequence restarts the stream.

**Comment by linknum23** [comment]
Hmm, even on the pi it is able to recover from that failure. I wonder if it recovers when launched from streams.py? If it doesn't I have an idea that the pipe's we are overflowing:
    
    self.proc = subprocess.Popen(args=inetradio_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=os.setpgrp)

Based on the code, it does not appear that the pipes have any real use. We don't use them to chain processes together. This should be investigated, and potentially removed or changed to log files like pianobar's implementation (stdin excluded):
   
    self.proc = subprocess.Popen(args='pianobar', stdin=subprocess.PIPE, stdout=open(pb_output_file, 'w'), stderr=open(pb_error_file, 'w'), env={'HOME' : pb_home})

**Comment by linknum23** [comment]
Tested this overnight since it occurs so infrequently (with the pipes removed). It still failed sometime in the night. There are a couple of different avenues to go from here:
1. Investigate whether or not vlc reports an error condition indicating the stream failed and restart vlc when it occurs
2. Investigate if there is an older version of vlc that doesn't have this bug. Based on some comments here: https://forum.videolan.org/viewtopic.php?t=152404, vlc version 3.07 may have less streaming issues.

We are still busy testing and finalizing the software for the **Early Bird** units so it will be a bit before we will have time to look at this.

**Comment by linknum23** [comment]
An initial fix for behavior like this was added in 0.1.8 by #297. Please reopen if this is not fixed.
